schedule/__init__.py

```diff
@@ -465,6 +486,9 @@ def _schedule_next_run(self):
             if (self.next_run - datetime.datetime.now()).days >= 7:
                 self.next_run -= self.period
 
+        if self.final_run and self.next_run > self.final_run:
+            self.next_run = datetime.datetime.max
```

Comment:
If we write this check in the `run()` method, we can return `CancelJob`, which in turn will make the scheduler unschedule the job all together. That way we don't need `datetime.datetime.max`.